hypothesis-python/src/hypothesis/configuration.py

```diff
@@ -57,14 +63,15 @@ def check_sideeffect_during_initialization(what: str, extra: str = "") -> None:
     elif _hypothesis_globals.in_initialization:
         # Note: -Werror is insufficient under pytest, as doesn't take effect until
         # test session start.
+        msg = what.format(*fmt_args)
         warnings.warn(
-            f"Slow code in plugin: avoid {what} at import time!  Set PYTHONWARNINGS=error "
+            f"Slow code in plugin: avoid {msg} at import time!  Set PYTHONWARNINGS=error "
             "to get a traceback and show which plugin is responsible." + extra,
             HypothesisSideeffectWarning,
             stacklevel=3,
         )
     else:
-        _first_postinit_what = what
+        _first_postinit_what = (what, fmt_args)
```

Comment:
Would it be better to stored the formatted `what`? Slightly concerned about holding onto a potentially mutable object.

Reply:
OTOH we know exactly what comes in, no user will call this function. Thanks for getting to all green :grin:

Reply:
Happy to help!